pinot-core/src/main/java/com/linkedin/pinot/core/data/recordtransformer/SanitationTransformer.java

```diff
@@ -19,50 +19,52 @@
 import com.linkedin.pinot.common.data.Schema;
 import com.linkedin.pinot.common.utils.StringUtil;
 import com.linkedin.pinot.core.data.GenericRow;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
 import java.util.Map;
 
 
 /**
  * The {@code SanitationTransformer} class will sanitize the values to follow certain rules including:
  * <ul>
  *   <li>No {@code null} characters in string values</li>
+ *   <li>Values are within the length limit</li>
  * </ul>
  * <p>NOTE: should put this after the {@link DataTypeTransformer} so that all values follow the data types in
  * {@link FieldSpec}.
  */
 public class SanitationTransformer implements RecordTransformer {
-  private final List<String> _stringColumns = new ArrayList<>();
+  private final Map<String, Integer> _stringColumnMaxLengthMap = new HashMap<>();
 
   public SanitationTransformer(Schema schema) {
     for (Map.Entry<String, FieldSpec> entry : schema.getFieldSpecMap().entrySet()) {
-      FieldSpec.DataType dataType = entry.getValue().getDataType();
-      if (dataType == FieldSpec.DataType.STRING) {
-        _stringColumns.add(entry.getKey());
+      FieldSpec fieldSpec = entry.getValue();
+      if (fieldSpec.getDataType() == FieldSpec.DataType.STRING) {
```

Comment:
may be useful to add a comment here that for now we sanitize only string types

Reply:
Added